src/ui/ChatterPostAttachment/ChatterPostAttachment.ts

```diff
@@ -49,6 +49,10 @@ export class ChatterPostAttachment extends Component {
 
       if (!Utils.isNullOrUndefined(Utils.getFieldValue(result, 'sfcontentfilename'))) {
         linkElement.text(Utils.getFieldValue(result, 'sfcontentfilename'));
+      } else if (!Utils.isNullOrUndefined(Utils.getFieldValue(result, 'sftitle'))) {
+        linkElement.text(Utils.getFieldValue(result, 'sftitle'));
+      } else if (!Utils.isNullOrUndefined(Utils.getFieldValue(result, 'sf_title'))) {
+        linkElement.text(Utils.getFieldValue(result, 'sf_title'));
```

Comment:
Why is it `sftitle` _or_ `sf_title` ? Why are both occurrences valid?

Reply:
Both are present in the mapping. The first is the Feed Attachment Title and the later the Generic Title.

Reply:
Code is getting harder to read, with multiple if/else if and lots of repetition.

Could be refactored to have a method named explicitely that helps determine the field to use on the result, and only one call to `linkElement.text`

It would also be important that it becomes explicit in the code that the order in which we check the "title" is actually important. Ideally just by naming methods properly, otherwise at least a comment that explains this.

Should also be enforced through a UT (as in : a test that check that if both value are present one take precedence over the other.)

Also, `sf_title` and `sftitle`, who decided that name ? :D Let's say it's a bit weak as far as a "good" naming goes to help understand which is which.

Reply:
Since the value that is changing is the second argument to `Utils.getFieldValue()`, you could create a method (as already suggested) maybe called `getSfTitleForResult(result) { <priority of lookup logic here>}`, and then usage would be:

```
const sfTitle = getSfTitleForResult(result);
linkElement.text(sfTitle);
```